src/Workspaces/CSharp/Portable/Editing/CSharpImportAdder.cs

```diff
@@ -57,5 +67,102 @@ private INamespaceSymbol GetExplicitNamespaceSymbol(ExpressionSyntax fullName, E
+                    return expanded.WithLeadingTrivia(leadingTrivia);
+                }
+
+                return node.WithLeadingTrivia(leadingTrivia);
+            }
+
+            public override SyntaxNode VisitGenericName(GenericNameSyntax node)
+            {
+                //Only visit leading trivia, as we only care about xml doc comments
+                var leadingTrivia = VisitList(node.GetLeadingTrivia());
+
+                if (_namespaceMembers.Contains(node.Identifier.Text))
+                {
+                    // no need to visit type argument list as simplifier will expand everything
+                    var expanded = Simplifier.Expand<SyntaxNode>(node, _model, _workspace, cancellationToken: _cancellationToken);
+                    return expanded.WithLeadingTrivia(leadingTrivia);
+                }
+
+                var typeArgumentList = (TypeArgumentListSyntax)base.Visit(node.TypeArgumentList);
+                return node.Update(node.Identifier.WithLeadingTrivia(leadingTrivia), typeArgumentList);
+            }
+
+            public override SyntaxNode VisitQualifiedName(QualifiedNameSyntax node)
+            {
+                var left = (NameSyntax)base.Visit(node.Left);
+                var right = node.Right;
+                if (right is GenericNameSyntax genericName)
+                {
+                    var typeArgumentList = (TypeArgumentListSyntax)base.Visit(genericName.TypeArgumentList);
+                    right = genericName.Update(genericName.Identifier, typeArgumentList);
+                }
+                return node.Update(left, node.DotToken, right);
+            }
+
+            public override SyntaxNode VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                // No need to visit trivia, as we only care about xml doc comments
+                if (node.Expression is MemberAccessExpressionSyntax memberAccess)
+                {
+                    if (_extensionMethods.Contains(memberAccess.Name.Identifier.Text))
+                    {
+                        // no need to visit this as simplifier will expand everything
+                        return Simplifier.Expand<SyntaxNode>(node, _model, _workspace, cancellationToken: _cancellationToken);
+                    }
+                }
+
+                return base.VisitInvocationExpression(node);
+            }
+
+            public override SyntaxNode VisitMemberAccessExpression(MemberAccessExpressionSyntax node)
+            {
+                node = (MemberAccessExpressionSyntax)base.VisitMemberAccessExpression(node);
+
+                if (_extensionMethods.Contains(node.Name.Identifier.Text))
+                {
+                    // we will not visit this if the parent node is expanded, since we just expand the entire parent node.
+                    // therefore, since this is visited, we haven't expanded, and so we should warn
```

Comment:
i don't really undestand the comment (though i think from our gitter conversation i know what this is about).  Can you have a full comment explaining the scenario for future readers?  Thanks!